tgui/packages/tgui/interfaces/MessageMonitor.tsx

```diff
@@ -133,7 +133,12 @@ const MessageLogsScreen = (props) => {
                 </Table.Cell>
                 <Table.Cell>{message.sender}</Table.Cell>
                 <Table.Cell>{message.recipient}</Table.Cell>
-                <Table.Cell>{message.message}</Table.Cell>
+                <Table.Cell>
+                  <Box
+                    as="span"
+                    dangerouslySetInnerHTML={{ __html: message.message }}
```

Comment:
what safeguards do we have against users leveraging this to input their own scripts inside tgui windows? just want to make sure we've thought of this in advance

Reply:
Absolutely. The emoji_parse call has been moved to the bottom of the sanitize message proc. This is called whenever any message from a pda is sent.
https://github.com/tgstation/tgstation/blob/da6b9b6e8e5416a8486bf43d350bd1626fd4c4ce/code/modules/modular_computers/file_system/programs/messenger/messenger_program.dm#L461-L471

As for non-messenger sources, any instances that might allow through unsanitized input, would already be vulnerable due to their direct output to tgchat (where emoji already show up)

I went through every source of PDA messages I could find to try and see how they handled sanitization

- Messenger app: sanitize_pda_message, sanitized
- mod pc circuits: goes through the app's send_signal function
- security officer joining: predefined message
- message monitor: tgui_input_text(), sanitized
- Request console (which uses the messaging server I guess?) html_encode, sanitized
- HUD glasses citations, tgui_input_text, sanitized
- warrant console citations: [uh oh](https://github.com/tgstation/tgstation/pull/84091)
- admin pda messages: not sanitized, presumably intentional